src/webln/NostrWeblnProvider.ts

```diff
@@ -213,6 +236,40 @@ export class NostrWebLNProvider {
       });
     });
   }
+
+  initNWC(provider: string, options: { name: string, returnTo?: string }) {
+    const height = 600;
+    const width = 400;
+    const top = window.outerHeight / 2 + window.screenY - height / 2;
+    const left = window.outerWidth / 2 + window.screenX - width / 2;
+    // @ts-ignore for my TS love :)
+    const url = new URL(NWC_URLS[provider]);
+    url.searchParams.set('c', options.name);
+    url.searchParams.set('p', this.publicKey);
+    url.searchParams.set('url', document.location.origin);
+    if (options.returnTo) {
+      url.searchParams.set('returnTo', options.returnTo);
+    }
+    return new Promise((resolve) => {
+      const popup = window.open(
+        url.toString(),
+        `${document.title} - Wallet Connect`,
+        `height=${height},width=${width},top=${top},left=${left}`
+      );
+      let processingResponse = false; // make sure we request the access token only once
```

Comment:
Would this be needed if you removed the event listener when you get a message?

Reply:
good point, I did not remove the handler